### ISSUE TITLE
Support for essential claims

### ISSUE THREAD

**Issue by praseodym** [mitreid-connect/OpenID-Connect-Java-Spring-Server#980]
The OIDC standard has support for marking claims as 'essential'; we should probably support this.

**Comment by jricher** [comment]
Agreed, we need deeper processing for the entire incoming claims object as in #473

**Comment by gazbert** [comment]
I'm willing to help out here if you don't mind me chipping away at a snail's pace - this is all new to me!

Perhaps we could deliver this functionality incrementally? Maybe start by supporting requests for only email scope claims in the id_token response - a bit like the Google Identity Server ID token response: https://developers.google.com/identity/protocols/OpenIDConnect

I've already started to play with the code and am currently returning some custom scoped claims in the ID token response for my use case.
